### PR TITLE
ci: update autoapprove workflow to v2 and pr target event

### PR DESCRIPTION
**Pull request by derberg**
**Description**

- recently autoapprove workflows were failing and this is the issue https://github.com/hmarr/auto-approve-action/issues/183 that explains what has changed to default GITHUB token

### CODE REVIEW

File: .github/workflows/automerge.yml [hunk 1]
```diff
@@ -17,10 +17,6 @@ on:
   pull_request_review:
     types:
       - submitted
-  check_suite:
```
Comment (derberg) [inline .github/workflows/automerge.yml:20]: I don't think we even needed to react to those events in our workflows

File: .github/workflows/automerge.yml [hunk 2]
```diff
@@ -29,7 +25,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Autoapproving
-        uses: hmarr/auto-approve-action@v2.0.0
+        uses: hmarr/auto-approve-action@v2
```
Comment (derberg) [inline .github/workflows/automerge.yml:28]: no we will not have to bump to every new 2.x version but always get latest improvements, line now we will also use 2.1 that improves logging of errors

Comment (magicmatatjahu) [reply .github/workflows/automerge.yml:28]: We don't need to tell like in npm `^v2` or something like that?

File: .github/workflows/automerge.yml [hunk 3]
```diff
@@ -4,7 +4,7 @@
 name: Automerge release bump PR
 
 on:
-  pull_request:
+  pull_request_target:
```
Comment (magicmatatjahu) [inline .github/workflows/automerge.yml:7]: Why do you wanna change it? I only ask 😄

File: .github/workflows/automerge.yml (hunk 2 above)
Comment (derberg) [reply .github/workflows/automerge.yml:28]: nope, it is all fetched from GitHub at the end, super manual process for action owner. You release for example v2.1.0 and then have to update release v2 again to match v2.1.0 🤷🏼‍♂️

File: .github/workflows/automerge.yml (hunk 3 above)
Comment (derberg) [reply .github/workflows/automerge.yml:7]: explanation is in linked issue. The problem is that in march github changed the scope of default GITHUB_TOKEN and it has read only rights on `pull_request` event, so the action fails as it has no rights to updat the status of the check through the API 😞 `pull_request_target` is relatively new event created mainly for GH Actions, so you basically mark that actions runs in scope of base not fork.

I'm a proud fighter for `pull_request_target` https://dev.to/derberg/github-actions-when-fascination-turns-into-disappointment-4d75 😄

Comment (magicmatatjahu) [reply .github/workflows/automerge.yml:7]: Thanks for explanation!

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**